server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -296,6 +297,29 @@ public class IndicesService extends AbstractLifecycleComponent
         Property.Dynamic
     );
 
+    public static final Setting<Integer> CLUSTER_DEFAULT_MAX_THREAD_COUNT_SETTING = new Setting<>(
+        "cluster.default.index.merge.scheduler.max_thread_count",
+        (s) -> Integer.toString(Math.max(1, Math.min(4, OpenSearchExecutors.allocatedProcessors(s) / 2))),
```

Comment:
nit - we can use similar approach of `MergeSchedulerConfig::getDefaultMaxThreadCount` here as well?